traitlets/traitlets.py

```diff
@@ -687,8 +690,6 @@ def __new__(cls, *args, **kw):
         inst._trait_values = {}
         inst._trait_notifiers = {}
         inst._cross_validation_lock = True
```

Comment:
These three lines are still `TraitType`-specific. We could either have a `pre_instance_init` method specialized in `HasTrait` or have `instance_init` of the trait type create these dictionaries if it is not there yet...